src/workerd/api/sql.h

```diff
@@ -22,9 +22,17 @@ class SqlStorage final: public jsg::Object, private SqliteDatabase::Regulator {
 
   class Cursor;
   class Statement;
+  struct SavepointResult {
+    bool success;
```

Comment:
Can we return whatever the callback returned, and in the case of an exception, propagate the exception? This would be consistent with the existing transaction API and I think is more natural. It would be too easy for someone to forget to check for `success` here and then when an exception happens it just gets swallowed.

Reply:
Yeah I agree, throwing is the right thing here. The point isn't to swallow errors, or even change how they are handled, it is to ensure that any writes are rolled back automatically. Have updated the code.